test/integration/kafka_sarama/kafka_test.go

```diff
@@ -21,16 +21,30 @@ import (
 )
 
 const (
-	TEST_GROUP_ID = "test_group_id"
+	TEST_GROUP_ID   = "test_group_id"
+	KAFKA_OFFSET    = "kafkaoffset"
+	KAFKA_PARTITION = "kafkapartition"
+	KAFKA_TOPIC     = "kafkatopic"
 )
 
+var TopicName = "test-ce-client-" + uuid.New().String()
+
 func TestSendEvent(t *testing.T) {
 	test.EachEvent(t, test.Events(), func(t *testing.T, eventIn event.Event) {
 		eventIn = test.ConvertEventExtensionsToString(t, eventIn)
 		clienttest.SendReceive(t, func() interface{} {
 			return protocolFactory(t)
 		}, eventIn, func(e event.Event) {
-			test.AssertEventEquals(t, eventIn, test.ConvertEventExtensionsToString(t, e))
+			got := test.ConvertEventExtensionsToString(t, e)
+
+			require.Equal(t, TopicName, got.Extensions()[KAFKA_TOPIC])
+			require.NotNil(t, got.Extensions()[KAFKA_PARTITION])
+			require.NotNil(t, got.Extensions()[KAFKA_OFFSET])
+
+			eventIn.SetExtension(KAFKA_OFFSET, got.Extensions()[KAFKA_OFFSET])
+			eventIn.SetExtension(KAFKA_PARTITION, got.Extensions()[KAFKA_PARTITION])
+			eventIn.SetExtension(KAFKA_TOPIC, TopicName)
+			test.AssertEventEquals(t, eventIn, got)
 		})
 	})
 }
```

Comment:
Suggesting the following changes for more clarity of the test (also for the other integration test):

```go
func TestSendEvent(t *testing.T) {
	test.EachEvent(t, test.Events(), func(t *testing.T, eventIn event.Event) {
		eventIn = test.ConvertEventExtensionsToString(t, eventIn)
		clienttest.SendReceive(t, func() interface{} {
			return protocolFactory(t)
		}, eventIn, func(eventOut event.Event) {
			eventOut = test.ConvertEventExtensionsToString(t, eventOut)

			require.Equal(t, TopicName, eventOut.Extensions()[KAFKA_TOPIC])
			require.NotNil(t, eventOut.Extensions()[KAFKA_PARTITION])
			require.NotNil(t, eventOut.Extensions()[KAFKA_OFFSET])

			test.AllOf(
				test.HasExactlyAttributesEqualTo(eventIn.Context),
				test.HasData(eventIn.Data()),
			)
		})
	})
}
```

Reply:
Renamed the variable names for more clarity and dropped extension comparison since we're not setting specific extensions here and have `require...` to assert on the specific extensions we want. Ideally we'd have a custom `test.EventMatcher` to assert on extensions but skip specific ones which get added at a later stage. I'll create tracking issue.